pandas/tests/arrays/categorical/test_indexing.py

```diff
@@ -145,3 +145,15 @@ def test_mask_with_boolean_raises(index):
 
     with pytest.raises(ValueError, match='NA / NaN'):
         s[idx]
+
+
+class NonCoercaibleCategorical(Categorical):
```

Comment:
I imagine this pattern would be generally useful to detect places where we accidentally convert an EA to an ndarray. What would be the best way to share this among tests? Would making a fixture that returned this be too weird? Monkeypatching?

Reply:
could do this  with a pytest monkeypatch (maybe just create an issue to follow up)